sdk/identity/azure-identity/test/ut/azure_cli_credential_test.cpp

```diff
@@ -344,6 +344,50 @@ TEST(AzureCliCredential, UnsafeChars)
   }
 }
 
+TEST(AzureCliCredential, UnsafeCharsScopeAndTenantId)
+{
+  std::string const ValidScopeButNotTenantId = "string:withcolon";
```

Comment:
I'd verify the whole set of 5 characters - first two accepted at both, last three are only allowed in scope. It can be a single unit test (no need for separate testcases), but I would use copy&paste to repeat some parts of the same test below, and not use a loop so that the failure message is clearer.

Reply:
I can fairly easily test for all the not allowed characters.

Validating allowed characters would still fail in CI for unrelated reasons, and I am going to leave out unit tests enhancements/refactoring as a separate work item, to avoid scope creep here.